File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size()==1) return(strs[0]);
        string res="";
        int m=INT_MAX;
        for (auto i=0;i<strs.size();i++){
            m=min(m,(int)strs[i].length());
        }
        bool t;
        for (auto i=0;i<m;i++){
            t=true;
            char c=strs[0][i];
            for (auto j=1;j<strs.size();j++){
                if (strs[j][i]!=c){ t=false; break;}
            }
            if (t) res+=c;
            else break;
        }
        return(res);
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
        def isAnagram(self, s: str, t: str) -> bool:
            if (len(s)!=len(t)): 
                return False
            dic={}
            for i in s:
                if (i not in dic ):
                    dic[i]=1
                else:
                    dic[i]+=1
            
            for j in t:
                if j in dic:
                    dic[j]-=1
            for i in dic:
                if dic[i]!=0:
                    return False
            return True
        def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
            dic={}
            for i in strs:
                if (tuple(sorted(i)) not in dic):
                    dic[tuple(sorted(i))]=[]
            for i in strs:
                dic[tuple(sorted(i))].append(i)
            res=[]
            for i in dic.values():
                res.append(i)
            return (res)



            